lib/_emerge/depgraph.py

```diff
@@ -9157,6 +9166,35 @@ def display_problems(self):
 			show_mask_docs()
 			writemsg("\n", noiselevel=-1)
 
+		features = self._frozen_config.settings.features
+		if "maintainer-nag" in features:
+			# Build a list of problematic packages
+			maintainer_nag_packages = []
+
+			for pkg in self._dynamic_config.digraph:
+				if(isinstance(pkg, Package) and pkg.repo == "gentoo"):
+					settings = pkg.root_config.settings
+					path = settings.environ()['PORTDIR'] + "/" + pkg.cp + "/metadata.xml"
```

Comment:
The PORTDIR variable is deprecated, so let's use `os.path.join(os.path.dirname(pkg.root_config.trees['porttree'].dbapi.findname(pkg.cpv, myrepo=pkg.repo)), 'metadata.xml')`.

Reply:
Thanks, that pointer is what I needed. Had a feeling it wasn't right!